### PR TITLE
Routes for rents

### PR DESCRIPTION
**Pull request by Gmarn**
# Summary

- Implementing 3 new routes (2 GET and 1 POST);
- Managing routes with the RentsController class;
- Some modifications on serializer to have the right result.

# Trello card

https://trello.com/c/ZS0KOSkc

### CODE REVIEW

File: app/controllers/api/v1/rents_controller.rb [hunk 1]
```diff
@@ -1,14 +1,27 @@
 module Api
   module V1
     class RentsController < ApiController
-      def index
-        @rent = Rent.all
-        render json: @rent
+      def users_rents
+        rent = Rent.where(user_id: params[:id])
```
Comment (Dinuuu) [inline app/controllers/api/v1/rents_controller.rb]: I would create a scope name `for_user(user)` for this, and use a Scope Policy

File: app/controllers/api/v1/rents_controller.rb [hunk 2]
```diff
@@ -1,14 +1,27 @@
...
+        rent = Rent.new(rent_params)
+        if rent.save
+          head :created
+        else
+          render json: { error: rent.errors.messages }, status: :bad_request
+        end
+      end
+
+      def books_rents
+        rent = Rent.where(book_id: params[:id])
```
Comment (Dinuuu) [inline app/controllers/api/v1/rents_controller.rb]: I would create a scope named `with_book(user)` for this

File: app/controllers/api/v1/rents_controller.rb (hunk 1 above)
Comment (Gmarn) [inline app/controllers/api/v1/rents_controller.rb]: Ok, but the scope policy using Pundit is for a card that comes next in the training. Any case i am going to do it here and now. Thanks

File: app/controllers/api/v1/rents_controller.rb [hunk 3]
```diff
@@ -0,0 +1,28 @@
+module Api
+  module V1
+    class RentsController < ApiController
+      def users_rents
+        rent = policy_scope(Rent)
+        render json: rent, each_serializer: UserRentSerializer
```
Comment (Gmarn) [inline app/controllers/api/v1/rents_controller.rb]: Because i need to add the book fields in the serializer

**Pull request merged**